File: android-frontend/SmartMedicine/view/appView/src/main/java/com/czy/appview/view/medicine/order/OrderViewPagerBar.java
package com.czy.appview.view.medicine.order;

import android.content.Context;
import android.util.AttributeSet;
import android.view.LayoutInflater;
import android.widget.LinearLayout;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.constraintlayout.widget.ConstraintLayout;
import androidx.core.content.ContextCompat;

import com.czy.appview.R;
import com.czy.appview.databinding.OrderViewpagerBarBinding;
import com.czy.appview.view.home.HomeViewPagerEnum;
import com.czy.domain.OnPositionItemClick;

public class OrderViewPagerBar extends ConstraintLayout {
    public OrderViewPagerBar(@NonNull Context context) {
        super(context);
        init(context);
    }

    public OrderViewPagerBar(@NonNull Context context, @Nullable AttributeSet attrs) {
        super(context, attrs);
        init(context);
    }

    public OrderViewPagerBar(@NonNull Context context, @Nullable AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        init(context);
    }

    private OrderViewpagerBarBinding binding;

    private void init(@NonNull Context context){
        LayoutInflater inflater = LayoutInflater.from(context);
        binding = OrderViewpagerBarBinding.inflate(inflater, this, true);

        setOnClickListener();
    }

    private void setOnClickListener(){
        LinearLayout[] linearLayouts = new LinearLayout[]{
                binding.lyClick1,
                binding.lyClick2
        };
        for (int i = 0; i < linearLayouts.length; i++){
            int finalI = i;
            linearLayouts[i].setOnClickListener(
                    v -> {
                        currentPosition = HomeViewPagerEnum.values()[finalI].getIndex();
                        if (onViewPagerBarClickListener != null){
                            onViewPagerBarClickListener.onPositionItemClick(
                                    HomeViewPagerEnum.values()[finalI].getIndex()
                            );
                            updateUI();
                        }
                    }
            );
        }
    }

    private int currentPosition = HomeViewPagerEnum.RECOMMEND.getIndex();

    private OnPositionItemClick onViewPagerBarClickListener;

    public void setOnViewPagerBarClickListener(@NonNull OnPositionItemClick onViewPagerBarClickListener) {
        this.onViewPagerBarClickListener = onViewPagerBarClickListener;
    }

    public void setCurrentPosition(int currentSelected){
        this.currentPosition = currentSelected;
        updateUI();
    }

    private void updateUI() {
        OrderViewPagerEnum homeViewPagerEnum = OrderViewPagerEnum.getByValue(currentPosition);
        binding.tvAppointment.setTextColor(
                OrderViewPagerEnum.APPOINTMENT_ORDER.equals(homeViewPagerEnum) ?
                        ContextCompat.getColor(getContext(), R.color.white) :
                        ContextCompat.getColor(getContext(), R.color.green_900)
        );
        binding.lyClick1.setBackgroundResource(
                OrderViewPagerEnum.APPOINTMENT_ORDER.equals(homeViewPagerEnum) ?
                        R.color.green_100 :
                        R.color.white
        );

        binding.tvPurchase.setTextColor(
                OrderViewPagerEnum.PURCHASE_ORDER.equals(homeViewPagerEnum) ?
                        ContextCompat.getColor(getContext(), R.color.white) :
                        ContextCompat.getColor(getContext(), R.color.green_900)
        );
        binding.lyClick2.setBackgroundResource(
                OrderViewPagerEnum.PURCHASE_ORDER.equals(homeViewPagerEnum) ?
                        R.color.green_100 :
                        R.color.white
        );
    }

}

File: android-frontend/SmartMedicine/view/appView/src/main/java/com/czy/appview/view/order/OrderViewPagerBar.kt
package com.czy.appview.view.order

import android.content.Context
import android.util.AttributeSet
import android.view.LayoutInflater
import androidx.constraintlayout.widget.ConstraintLayout
import androidx.core.content.ContextCompat
import com.czy.appview.R
import com.czy.appview.databinding.OrderViewpagerBarBinding
import com.czy.domain.OnPositionItemClick

class OrderViewPagerBar : ConstraintLayout {

    private var binding: OrderViewpagerBarBinding
    private var currentPosition: Int = OrderViewPagerEnum.APPOINTMENT_ORDER.value
    private var onViewPagerBarClickListener: OnPositionItemClick? = null

    constructor(context: Context) : super(context) {
        binding = OrderViewpagerBarBinding.inflate(LayoutInflater.from(context), this, true)
        init()
    }

    constructor(context: Context, attrs: AttributeSet?) : super(context, attrs) {
        binding = OrderViewpagerBarBinding.inflate(LayoutInflater.from(context), this, true)
        init()
    }

    constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int) : super(context, attrs, defStyleAttr) {
        binding = OrderViewpagerBarBinding.inflate(LayoutInflater.from(context), this, true)
        init()
    }

    private fun init() {
        setOnClickListener()
    }

    private fun setOnClickListener() {
        val linearLayouts = arrayOf(binding.lyClick1, binding.lyClick2)
        for (i in linearLayouts.indices) {
            linearLayouts[i].setOnClickListener {
                currentPosition = OrderViewPagerEnum.values()[i].value
                onViewPagerBarClickListener?.onPositionItemClick(OrderViewPagerEnum.values()[i].value)
                updateUI()
            }
        }
    }

    private fun updateUI() {
        val medicineViewPagerEnum = OrderViewPagerEnum.getByValue(currentPosition)

        binding.tvAppointment.setTextColor(
            if (OrderViewPagerEnum.APPOINTMENT_ORDER == medicineViewPagerEnum) {
                ContextCompat.getColor(context, R.color.green_1000)
            }
            else {
                ContextCompat.getColor(context, R.color.green_900)
            }
        )
        binding.vBar1.visibility = if (OrderViewPagerEnum.APPOINTMENT_ORDER == medicineViewPagerEnum)
            VISIBLE else GONE

        binding.tvPurchase.setTextColor(
            if (OrderViewPagerEnum.PURCHASE_ORDER == medicineViewPagerEnum) {
                ContextCompat.getColor(context, R.color.green_1000)
            }
            else {
                ContextCompat.getColor(context, R.color.green_900)
            }
        )
        binding.vBar2.visibility = if (OrderViewPagerEnum.PURCHASE_ORDER == medicineViewPagerEnum)
            VISIBLE else GONE
    }

}